services/ymax-planner/src/kv-store.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * Key-value store interface for storing and retrieving configuration values.
+ * This abstraction allows for different storage backends to be plugged in.
+ */
+import Database from 'better-sqlite3';
+
+export interface KeyValueStore {
+  /**
+   * Get a value from the store by key
+   * @param key - The key to retrieve
+   * @returns The value if found, undefined otherwise
+   */
+  get(key: string): Promise<string | undefined>;
+
+  /**
+   * Set a value in the store
+   * @param key - The key to set
+   * @param value - The value to store
+   */
+  set(key: string, value: string): Promise<void>;
+
+  /**
+   * Delete a value from the store
+   * @param key - The key to delete
+   */
+  delete(key: string): Promise<void>;
+
+  /**
+   * Check if a key exists in the store
+   * @param key - The key to check
+   */
+  has(key: string): Promise<boolean>;
+}
+
+export type SQLiteKeyValueStoreConfig = {
+  dbPath: string;
+  tableName?: string;
+};
+
+/**
+ * SQLite implementation of KeyValueStore.
+ * Stores key-value pairs in a SQLite database file.
+ * Useful for production environments with persistent local storage.
+ *
+ * Note: this implementation uses better-sqlite which is synchronous (https://github.com/WiseLibs/better-sqlite3)
+ * but the functions still use `async` since they implement the KeyValueStore interface
+ *
+ * The table will be created automatically if it doesn't exist with the following schema:
+ * - key_name: TEXT PRIMARY KEY
+ * - key_value: TEXT NOT NULL
+ * - updated_at: INTEGER (timestamp in milliseconds)
+ *
+ * Usage:
+ * ```typescript
+ * const kvStore = makeSQLiteKeyValueStore({
+ *   dbPath: './data/kv-store.db',
+ *   tableName: 'kv_store', // optional, defaults to 'kv_store'
+ * });
+ * ```
+ */
+export const makeSQLiteKeyValueStore = (
+  config: SQLiteKeyValueStoreConfig,
+): KeyValueStore => {
```

Comment:
Since the path is always and obviously going to be required, I think it should be promoted up from the options back to an independent parameter.
```suggestion
export const makeSQLiteKeyValueStore = (
  dbPath: string,
  options: SQLiteKeyValueStoreOptions,
): KeyValueStore => {
```